src/NuGet.Core/NuGet.Common/Migrations/MigrationRunner.cs

```diff
@@ -34,24 +24,21 @@ public static void Run()
                 {
                     if (WaitForMutex(mutex))
                     {
-                        if (!File.Exists(expectedMigrationFilename))
+                        try
                         {
                             // Only run migrations that have not already been run
-                            int highestMigrationRun = GetHighestMigrationRun(migrationsDirectory);
-                            for (int i = highestMigrationRun + 1; i < Migrations.Count; i++)
+                            if (!File.Exists(expectedMigrationFilename))
                             {
-                                try
-                                {
-                                    Migrations[i]();
-                                    // Create file for every migration run, so that if an older version of NuGet is run, it doesn't try to run
-                                    // migrations again.
-                                    string migrationFile = Path.Combine(migrationsDirectory, (i + 1).ToString(CultureInfo.InvariantCulture));
-                                    File.WriteAllText(migrationFile, string.Empty);
-                                }
-                                catch { }
+                                Migration1.Run();
+                                // Create file for the migration run, so that if an older version of NuGet is run, it doesn't try to run migrations again.
+                                File.Create(expectedMigrationFilename).Dispose();
                             }
                         }
-                        mutex.ReleaseMutex();
+                        catch { }
```

Comment:
What happens if Migration1.Run() succeeds but then for some reason `File.Create(...).Dispose()` fails?  I assume it's ok that subsequent invocations will try to migrate again?

Reply:
> I assume it's ok that subsequent invocations will try to migrate again?

yes, that is correct. The subsequent invocation will check for the existence of the migration file, and it does not exist in that case and hence NuGet runs migration again.